npm/npm-debug-cli/dataplaneConverter/converter_test.go

```diff
@@ -0,0 +1,220 @@
+package converter
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/npm/npm-debug-cli/dataplaneParser/iptable"
+	"github.com/Azure/azure-container-networking/npm/npm-debug-cli/pb"
+)
+
+func TestGetJSONRulesFromIptable(t *testing.T) {
+	var (
+		tableName = "filter"
+	)
+
+	c := &Converter{}
+	c.GetJSONRulesFromIptable(tableName, "../testFiles/npmCache.json", "../testFiles/clusterIptableSave")
+}
+
+func TestGetProtobufRulesFromIptable(t *testing.T) {
+	var (
+		tableName = "filter"
+	)
+
+	c := &Converter{}
+	c.GetProtobufRulesFromIptable(tableName, "../testFiles/npmCache.json", "../testFiles/clusterIptableSave")
+}
+
+func TestGetSetType(t *testing.T) {
+	type test struct {
+		inputSetName string
+		inputMapName string
+		expected     pb.SetType
+	}
+
+	t0 := &test{inputSetName: "ns-testnamespace", inputMapName: "SetMap", expected: pb.SetType_NAMESPACE}
+	t1 := &test{inputSetName: "app:frontend", inputMapName: "SetMap", expected: pb.SetType_KEYVALUELABELOFPOD}
+	t2 := &test{inputSetName: "k1:v0:v1", inputMapName: "ListMap", expected: pb.SetType_NESTEDLABELOFPOD}
+	t3 := &test{inputSetName: "all-namespaces", inputMapName: "ListMap", expected: pb.SetType_KEYLABELOFNAMESPACE}
+	t4 := &test{inputSetName: "namedport:serve-80", inputMapName: "SetMap", expected: pb.SetType_NAMEDPORTS}
+	t5 := &test{inputSetName: "k0", inputMapName: "SetMap", expected: pb.SetType_KEYLABELOFPOD}
+	t6 := &test{inputSetName: "ns-namespace:test0", inputMapName: "ListMap", expected: pb.SetType_KEYVALUELABELOFNAMESPACE}
```

Comment:
this content is good, but the format is unusual. you can turn this in to typical table-driven test format really easily though:
```
var tests := struct {
	inputSetName string
	inputMapName string
	expected     pb.SetType
}{
	{inputSetName: "ns-testnamespace", inputMapName: "SetMap", expected: pb.SetType_NAMESPACE},
	...
}
```
and then when you iterate your tests, you might want to run them as subtests using `t.Run`.
[more testing info here](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)